Fixes #5168

SheepRock opened the pull request:
## Summary

Corrected the link to the page containing the C# 6.0 index-based syntax initializer.

Fixes #5168

SheepRock committed:
Fixes #5168

pkulikov commented on docs/csharp/whats-new/csharp-version-history.md:
```diff
@@ -94,7 +94,7 @@ With versions 3.0 and 5.0, C# had added some impressive features in an object or
 - [String interpolation](../language-reference/tokens/interpolated.md)
 - [nameof operator](https://stackoverflow.com/questions/31695900/what-is-the-purpose-of-nameof)
-- [Dictionary initializer](../programming-guide/classes-and-structs/how-to-initialize-a-dictionary-with-a-collection-initializer.md)
+- [Dictionary initializer](csharp-6.md#index-initializers)
```
@SheepRock Thank you for fixing that! To make the whole link correct, I suggest to change the link text as well to "Index initializers* as initializing dictionary is a specific use case of applying index initializers.

SheepRock committed:
Fixes the feature name

Changes from Dictionary initializer to Index initializers, as suggest by @pkulikov

BillWagner reviewed (commented):
Thanks for spotting this and fixing it @SheepRock 

We appreciate it. I've reviewed the changes, and I'll :shipit: now.

You should see the changes in the next day or two on our regular publish cycle.

Thanks again.
